src/test/java/org/jenkinsci/plugins/scriptsecurity/scripts/ScriptApprovalTest.java

```diff
@@ -136,6 +143,22 @@ public void malformedScriptApproval() throws Exception {
         assertEquals(0, sa.getDangerousApprovedSignatures().length);
     }
 
+    @Issue("JENKINS-57563")
+    @LocalData("upgrade")
```

Comment:
FWIW if you rename the zip file to `upgradeSmokes` to match the name of the test then you don't need to specify a name here. I might also add a small note like: `// Just a scriptApproval.xml that whitelists 'staticMethod jenkins.model.Jenkins getInstance'`

Reply:
You should be able to use the xml by it self instead of using a zip file
See: https://github.com/jenkinsci/matrix-auth-plugin/pull/48/files for comparison

Reply:
Yeah good point, for a simple case like this just having the file specified directly might be easier.

Reply:
I've tried what @dwnusbaum suggested, it works as expected, so I'm including it in my last set of commits. 
I've also tried to deal with the approach advised by @casz, but I had to do sth wrong. I placed the file with the same content as in the ZIP into resources/.../ScriptApprovalTest/security/config.xml, but it doesn't work. I've tried several other changes, but nothing works for me. If we really need to remove the 'ZIP approach', could you advise me where exactly and which content should be placed? In that case we should change also other tests in ScriptApprovalTest to unify that, thank you.

Reply:
@pjanouse The zip approach is fine, it has the same behavior as what @casz is recommending, the benefit is just that it would be easier for anyone browsing the source code to understand the test without needing to unzip the archive. I think the resource path you would want would be `src/main/resources/.../ScriptApprovalTest/upgradeSmokes/scriptApproval.xml`, i.e. you just unzip your archive into a directory with the same name. If that isn't working for you, no worries, we can leave it as is.

Reply:
Hmm, I think this was one of my un-successful attempt... tried again and it works as expected, so using...